lib/intMap.js

```diff
@@ -0,0 +1,51 @@
+var hasOwn = Object.prototype.hasOwnProperty;
+
+/**
+ * A very simple polyfill for Map with limited functionality
+ * that only works for integer keys.
+ */
+function IntMap() {
+    this.map = {};
```

Comment:
Not sure if there's much point but you could reuse the isES5 in `index.js` function to see whether you could use `Object.create(null)` here.